jvm-slides/src/main/groovy/org/asciidoctor/gradle/jvm/slides/RevealJSOptions.groovy

```diff
@@ -109,10 +109,18 @@ class RevealJSOptions {
         this.progressBar = Optional.ofNullable(b)
     }
 
-    /** Display the slide number of the current slide.
-     *
+    /**
+     * Display the slider number of the current slide. <br/>
+     * true will display the slide number with default formatting. <br/>
+     * Additional formatting is available:
+     * <ul>
+     *     <li>h.v: horizontal . vertical slide number (default)</li>
+     *     <li>h/v: horizontal / vertical slide number</li>
+     *     <li>c: flattened slide number</li>
+     *     <li>c/t: flattened slide number / total slides</li>
+     * </ul>
      */
-    void setSlideNumber(Boolean b) {
+    void setSlideNumber(String b) {
```

Comment:
Completely replacing it this way might come as a surprise to some users, so I would suggest keeping the method that takes a Boolean, and mark it as `@Deprecated`. Also add a `logger.warn` inside it, so tell people to use the new method. Then forward the method to your   new one, where `true` would pass `c` and `false` passes `null`.

Reply:
Also I am wondering whether we should rather use an enum here, where the values of the enum matched `c`, `c/t` etc.

Reply:
> Completely replacing it this way might come as a surprise to some users, so I would suggest keeping the method that takes a Boolean, and mark it as @Deprecated

I wasn't sure it was allowed to have two methods with the same name (but a different argument type).

> Also I am wondering whether we should rather use an enum here, where the values of the enum matched c, c/t etc.

I thought about it but I think it would be limiting and it will create a tight coupling.
On the plus side, it will be more readable/safe.

```gradle
sliderNumber = SlideNumber.FLATTENED // instead of 'c'
```

Maybe we should provide both? Is that allowed?

Reply:
Both is allowed. You can actually do more. Have a look at how setSafeMode is implemented https://github.com/asciidoctor/asciidoctor-gradle-plugin/blob/63c508caffb7fb6a3331732f3c7c8e00005c44fd/asciidoctor-gradle-jvm/src/main/groovy/org/asciidoctor/gradle/jvm/AsciidoctorJExtension.groovy#L501

Reply:
Nice!
In that case, we should add an enum and mark `void setSliderNumber(Boolean b)` as deprecated.

Reply:
Use `SafeMode` as an archetype and you should be good.

Reply:
Any suggestions on the naming?

**concise**
```groovy
enum SlideNumber {
  DEFAULT('h.v'),
  HORIZONTAL_VERTICAL('h/v'),
  COUNT('c'),
  COUNT_TOTAL('c/t')
}
```

**descriptive**
```groovy
enum SlideNumber {
  HORIZONTAL_DOT_VERTICAL('h.v'),
  HORIZONTAL_SLASH_VERTICAL('h/v'),
  FLATTENED_COUNT('c'),
  FLATTENED_COUNT_SLASH_TOTAL('c/t')
}
```